### PR TITLE
PR #3 : Creating Login Functionality

### PR DESCRIPTION
**Pull request by Rapter1990**
Implement the proces of Login User

### CODE REVIEW

File: crm-base-api/src/controllers/UsersController.ts [hunk 1]
```diff
@@ -56,4 +73,50 @@ export class UsersController {
...
+                {
+                    userId: user.userId,
+                    username: user.username,
+                    email: user.email,
+                },
+                process.env.JWT_SECRET || 'default_secret', // Use environment variable or fallback to default
+                { expiresIn: '1d' }
+            );
+
+            // Send token to client
```
Comment (EmreKiziltoprak) [inline crm-base-api/src/controllers/UsersController.ts:114]: response should be added with generic ApiResponse. Manuel response must be deleted and Custom Class usage should be maintained.

File: crm-base-api/src/test/integration/users.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                username: 'testuser',
+                email: 'test@example.com',
+                password: 'password123',
+            };
+
+            // Mocking userService.findByEmail to return null (user doesn't exist)
+            jest.spyOn(usersService, 'findByEmail').mockResolvedValue(null);
+
+            // Mocking userService.createUser to return a user
+            jest.spyOn(usersService, 'createUser').mockResolvedValue({
```
Comment (EmreKiziltoprak) [inline crm-base-api/src/test/integration/users.controller.spec.ts:60]: automatic id test should be added

File: crm-base-api/src/controllers/UsersController.ts [hunk 3]
```diff
@@ -56,4 +73,50 @@ export class UsersController {
...
+
+    /**
+     * Logs in a user.
+     * @param {LoginRequest} body - The request body containing user login information.
+     * @param {Response} res - The response object for sending HTTP responses.
+     * @returns {Promise<any>} A promise resolving to the authentication token if login is successful.
+     * @throws {UnauthorizedError} Throws an unauthorized error if the provided credentials are invalid.
+     * @throws {HttpError} Throws an HTTP error if an unexpected error occurs during login.
+     */
+    @Post('/login')
```
Comment (EmreKiziltoprak) [inline crm-base-api/src/controllers/UsersController.ts:85]: userId should be generated automatically.

File: crm-base-api/src/controllers/UsersController.ts (hunk 1 above)
Comment (EmreKiziltoprak) [reply crm-base-api/src/controllers/UsersController.ts:114]: within the response when token is generated success code should be provided.

Comment (EmreKiziltoprak) [reply crm-base-api/src/controllers/UsersController.ts:114]: when there is error it should be provided too.

**Review by EmreKiziltoprak — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Print directory structure for debugging
+      run: ls -R
+
+    - name: Print current directory
```
[the remark this answers is not in the document]
Comment (Rapter1990) [reply .github/workflows/ci.yml:23]: It's resolved.

**Review by EmreKiziltoprak — APPROVED** [review]
Changes are registered, and required attachments are created.

**Pull request merged**